feat: migrate launch template to use SSM for AMI lookup

npalm opened the pull request:
# Description

This PR migrates the launch template to use SSM to foor looking up AMI ID. This PR is inteded to migrate to use SSM for AMI lookups. In later PR we will remove logic from scale-up and remove old parateters.

By default, the module still uses the same AMI filters, but difference is the resulting AMI ID is stored in SSM instead of the launch template. As alternative the module user can provide `ami_id_ssm_parameter_arn` pointing to an SSM parameter containing the AMI ID to be used (data type aws:ec2:image). 

When using a custom managed AMI this will avoid drift in terraform when `ami.id_ssm_parameter_arn` is used instead of the deprecated `ami_id_ssm_parameter_name`. Another way to avoid the drift is to ensure the filter is poiting to the same image. See also #4460

In PR #4488 and updater via Lambda is introduced. I think we should firt think how we can better support the AMI id in the core. What this PR is supposed todo. NExt the update can be an nice way to auto update AMI's without running terraform.

The PR also refactor the variables for `ami_*` and move the to a dedicated object `ami`. The old ones remains till the next major release. Deprecattion warnings are added to the output.

## Notes

- Adjuest multi runner examples
- Test with multi runners
- Test with default runners + pool

## Fix

- Deprecation warning inline_poilicy for pool

dgokcin commented on modules/runners/main.tf:
```diff
@@ -140,7 +162,7 @@ resource "aws_launch_template" "runner" {
 
   instance_initiated_shutdown_behavior = "terminate"
-  image_id                             = data.aws_ami.runner.id
+  image_id                             = "resolve:ssm:${local.ami_id_ssm_module_managed ? aws_ssm_parameter.runner_ami_id[0].arn : var.ami_id_ssm_parameter_arn}"
```
is there a reason behind using an arn instead of the parameter name 🤔

npalm replied:
The only thought it would be more specific. I just pushed an example that works with a public parameter as well.

npalm replied:
Do you think a name will works as well. I thought using the arn makes it also simple to setup IAM policies.

npalm replied:
I did inentially intoduce a new parameter, since the SSM parameter requires the correct data type. The old parameter is handled by the lambda. The new one fully by the launch template. I want to get rid of the old one in a breaking change release.

npalm commented:
Updated:

- Add to multi runner example, the usages of a public aws managed paramater.
- Added deprecation output

Some more todo work on my list
- Crack my had on introducing an object for AMI related settings.
- Update docs

@dgokcin thanks for already having a look. Would you have time to review the PR once I done with the draft. Early feedback is always welcome.

npalm commented:
@dgokcin would you have time to test hte PR?

dgokcin commented:
> @dgokcin would you have time to test hte PR?

can try it on monday :)

npalm commented:
> > @dgokcin would you have time to test hte PR?
> 
> can try it on monday :)

did you had time to run a quikc check?

dgokcin reviewed (approved):
it looks good to me! I think this is a good step towards my initial problem with drift in ami names!
